core/src/solve.jl

```diff
@@ -787,12 +814,23 @@ function formulate_flow!(
             continue
         end
 
-        # For now allocated = demand
-        for priority in eachindex(allocated[i])
-            allocated[i][priority] = demand[i][priority](t)
-        end
+        q = 0.0
 
-        q = sum(allocated[i])
+        if allocation_optimized[i]
```

Comment:
I wonder if introducing `allocation_optimized` is worth it. Would it also be sufficient to initialize `allocated` to `Inf`, and then take the minimum of that and the demand in either case?